packagehandlers/mavenpackagehandler.go

```diff
@@ -153,15 +153,29 @@ type MavenPackageHandler struct {
 }
 
 func (mph *MavenPackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
-	if err := mph.getProjectPoms(); err != nil {
+	depTreeOutput, clearMavenDepTreeRun, err := mph.runMavenDepTreeIfNeeded()
+	if err != nil {
 		return err
 	}
+
+	err = mph.getProjectPoms(depTreeOutput)
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	depTreeOutput, clearMavenDepTreeRun, err := mph.runMavenDepTreeIfNeeded()
	if err != nil {
		return err
	}

	err = mph.getProjectPoms(depTreeOutput)
	if err != nil {
		return err
	}
```
this those two should be together on the same func